src/test/scala/chiselTests/experimental/FlatIOSpec.scala

```diff
@@ -76,4 +79,44 @@ class FlatIOSpec extends ChiselFlatSpec {
     chirrtl should include("output a : UInt<1>")
     chirrtl should include("output b : UInt<2>[2]")
   }
+
+  it should "maintain port order for Bundles" in {
+    class MyBundle extends Bundle {
+      val foo = Bool()
+      val bar = Bool()
+    }
+    class MyModule extends Module {
+      val io = IO(Input(new MyBundle))
+    }
+    class MyFlatIOModule extends Module {
+      val io = FlatIO(Input(new MyBundle))
+    }
+
+    matchesAndOmits(
+      ChiselStage.emitSystemVerilog(new MyModule)
+    )("io_foo,")("io_bar,")
+
+    matchesAndOmits(
+      ChiselStage.emitSystemVerilog(new MyFlatIOModule)
+    )("foo,")("bar,")
+  }
```

Comment:
Clever, I need to just finish and merge https://github.com/chipsalliance/chisel/pull/3410 as that would be a good way to test this.

I wonder if we should also check Vec and also check recursive order...